src/tls/cockpit-certificate-ensure.c

```diff
@@ -277,12 +284,20 @@ certificate_and_key_split (CertificateKeyPair *self)
   return false;
 }
 
-static void
+static int
 certificate_and_key_read (CertificateKeyPair *self,
                           const char         *certificate_filename)
 {
   self->certificate_filename = strdupx (certificate_filename);
-  read_file (self->certificate_filename, &self->certificate);
+
+  if (read_file (self->certificate_filename, &self->certificate) < 0)
+    return -1;
+
+  if (self->certificate.size == 0)
```

Comment:
This  feels arbitrary and too brittle. If the concern is short writes after a power loss, then "0 bytes" feels like a "lucky unlucky" case.

The normal way to avoid that is to write the file with a temp name, fsync() it, and then rename it to the final name. Your third commit aims at that, I'll comment on that separately.

On the reading side, the gnutls functions to parse the certificate should already fail (in `certificate_and_key_parse_to_creds()`) if the certificate is empty, or if the file is truncated, so this check here ought to be redundant?